src/plugins/PluginAPI.hpp

```diff
@@ -59,6 +59,9 @@ struct SVersionInfo {
 #define OPTIONAL
 #define HANDLE void*
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreturn-type-c-linkage"
```

Comment:
This actually works for clang too despite its name https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas

Reply:
even if it didn't, it'd likely be ignored